### PR TITLE
docs: reorder sdk docs

### PR DESCRIPTION
**Pull request by vinckr**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: docs/sdk.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+:::
+
+:::caution old sdks
+
+[`hydra-client`](https://github.com/ory/sdk/tree/master/clients/hydra),
+[`kratos-client`](https://github.com/ory/sdk/tree/master/clients/kratos), and
+[`keto-client`](https://github.com/ory/sdk/tree/master/clients/keto) SDKs are no longer actively maintained.  
+Read the [Ory SDK upgrade guide](https://www.ory.sh/docs/guides/upgrade/sdk-v1) to learn more.
+
+:::
```
Comment (aeneasr) [inline docs/sdk.mdx:50]: I think we'll keep these SDKs around for a while

```suggestion
:::caution Self-hosted SDKs

Ory releases SDKs which can more easily be used when self-hosting Ory software. These SDKs follow the open source release schedule. These SDKs are available for [`hydra-client`](https://github.com/ory/sdk/tree/master/clients/hydra),
[`kratos-client`](https://github.com/ory/sdk/tree/master/clients/kratos), and
[`keto-client`](https://github.com/ory/sdk/tree/master/clients/keto).

:::
```

Comment (vinckr) [reply docs/sdk.mdx:50]: So it is recommended to use the "old" sdk when selfhosting? 

bc earlier it says

>It (the Ory "client" SDK) works with Ory Network and when self-hosting Ory services.

Comment (aeneasr) [reply docs/sdk.mdx:50]: It's a bit ambigous. Both SDKs work with both systems. But the ory client SDK follows the release steps of the Ory Network, and the other SDKs follow the release steps of the self-hosted stuff. That's the difference. So if you use ory client v1.2.3 then you don't know what version of Ory Kratos that is compatible with, for example.

Comment (vinckr) [reply docs/sdk.mdx:50]: Gotcha, 
then I would definitely recommend to use the kratos-client etc. SDK when selfhosting
and you have to use client for Ory Network (pointing out the difference)
and have a dedicated SDK page for kratos etc. in the selfhosting reference section.

putting this back in draft state until I have done the above and ask for review again in a bit.

**Comment by vinckr** [comment]
changed it so it has a main SDK doc and one for each project. 
Also new format + text, and added a note that hopefully explains which SDK to use when.

File: docs/hydra/sdk/01_overview.md [hunk 2]
```diff
@@ -1,40 +1,62 @@
...
 ---
 
-All SDKs use automated code generation provided by [`openapi-generator`](https://github.com/OpenAPITools/openapi-generator).
-Unfortunately, `openapi-generator` has serious breaking changes in the generated code when upgrading versions. Therefore, we don't
-make backwards compatibility promises with regards to the generated SDKs. We hope to improve this process in the future.
+The Ory Hydra SDK allows for integration with a self-hosted [Ory Hydra OAuth2 Server](https://github.com/ory/hydra).
 
-Before you check out the SDKs, head over to the [REST API](reference/api.mdx) documentation which includes code samples for common
-programming languages for each REST endpoint.
+Before using the SDK, it is recommended to consult the [Ory REST API](../reference/api) documentation, which includes code samples
```
Comment (tomekpapiernik) [inline docs/hydra/sdk/01_overview.md:9]: Please use active voice and imperative mood. 
The way we're linking to docs now must use filenames, so please revert to `api.mdx`.
Also, are there actual code samples/examples for various programming languages in the API reference?

**Review by tomekpapiernik — CHANGES_REQUESTED** [review]
The docs you edited have a similar structure so I didn't duplicate my comments. Make sure to address the highlighted issues in all docs in this PR. Thanks!

File: src/sidebar.js [hunk 3]
```diff
@@ -609,30 +608,27 @@ module.exports = {
...
         },
-      ],
-    },
-    {
-      "Ory Keto": [
         {
-          type: "autogenerated",
-          dirName: "keto/sdk",
+          type: "category",
+          label: "Ory OAuth2",
```
[the remark this answers is not in the document]
Comment (vinckr) [reply src/sidebar.js:628]: you mean Ory OAuth2 & OpenID, right?

File: docs/hydra/sdk/01_overview.md (hunk 2 above)
Comment (vinckr) [reply docs/hydra/sdk/01_overview.md:9]: done, 

as for the code samples in some cases there is someting like 

```js 
pseudo-code example 
router.get('/recovery', async function (req, res) { const flow = await client.getVerificationFlow(req.header('cookie'), req.query['flow'])
res.render('verification', flow) })
```

It does contain explanations on every REST endpoint. let me know if you can think of a better phrasing

**Comment by vinckr** [comment]
Hey @tomekpapiernik 
I changed the formatting, so only one admonition left. 
Let me know if you can think of a better heading than "Further information". 
Addressed your other comments as well :) 
Basically 3 things to be conveyed in that section:

- the link to the source code
- the call to action to create an issue if you are missing your language
- the note which SDK to use for Ory self-hosted & Network, vice versa

also the Hydra SDK has a some extra info on the oauth2 endpoints.

**Comment by aeneasr** [comment]
This looks good so far to me! The SDK examples themselves need to be reworked, but we can do that in another PR :)

**Review by tomekpapiernik — APPROVED** [review]

**Pull request merged**